testing/test_junitxml.py

```diff
@@ -61,79 +63,88 @@ def run_and_parse(pytester: Pytester, schema: xmlschema.XMLSchema) -> RunAndPars
     return RunAndParse(pytester, schema)
 
 
-def assert_attr(node, **kwargs):
+def assert_attr(node: minidom.Element, **kwargs: object) -> None:
     __tracebackhide__ = True
 
-    def nodeval(node, name):
+    def nodeval(node: minidom.Element, name: str) -> str | None:
         anode = node.getAttributeNode(name)
-        if anode is not None:
-            return anode.value
+        return anode.value if anode is not None else None
 
     expected = {name: str(value) for name, value in kwargs.items()}
     on_node = {name: nodeval(node, name) for name in expected}
     assert on_node == expected
 
 
 class DomNode:
-    def __init__(self, dom):
+    def __init__(self, dom: minidom.Element | minidom.Document):
         self.__node = dom
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self.__node.toxml()
 
-    def find_first_by_tag(self, tag):
+    def find_first_by_tag(self, tag: str) -> DomNode | None:
         return self.find_nth_by_tag(tag, 0)
 
-    def _by_tag(self, tag):
-        return self.__node.getElementsByTagName(tag)
-
     @property
-    def children(self):
+    def children(self) -> list[DomNode]:
         return [type(self)(x) for x in self.__node.childNodes]
 
     @property
-    def get_unique_child(self):
+    def get_unique_child(self) -> DomNode:
         children = self.children
         assert len(children) == 1
         return children[0]
 
-    def find_nth_by_tag(self, tag, n):
-        items = self._by_tag(tag)
+    def find_nth_by_tag(self, tag: str, n: int) -> DomNode | None:
+        items = self.__node.getElementsByTagName(tag)
         try:
             nth = items[n]
         except IndexError:
-            pass
+            return None
         else:
             return type(self)(nth)
 
-    def find_by_tag(self, tag):
+    def find_by_tag(self, tag: str) -> list[DomNode]:
         t = type(self)
         return [t(x) for x in self.__node.getElementsByTagName(tag)]
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: str) -> str:
+        if isinstance(self.__node, minidom.Document):
+            raise TypeError(type(self.__node))
```

Comment:
What's the goal of this? Maybe needs a better message?

Reply:
good catch - it needs to note, that based on [Document objects](https://docs.python.org/3/library/xml.dom.html#dom-document-objects) items are only avaliable on the `Document.documentElement`

Reply:
i think i'll create a DomRoot wrapper to make it easy